modules/FindOVR.cmake

```diff
@@ -39,28 +39,61 @@
 #   DEALINGS IN THE SOFTWARE.
 #
 
-set(LIBOVR_ROOT ${CMAKE_INSTALL_PREFIX})
-
 if(NOT OVR_SDK_ROOT)
-    message(WARNING "OVR_SDK_ROOT is not set. Will try to find headers and library in ${CMAKE_INSTALL_PREFIX}." )
-else()
-    set(LIBOVR_ROOT ${OVR_SDK_ROOT}/LibOVR)
+   find_path(OVR_SDK_ROOT OculusSDK)
+endif()
+
+if(OVR_SDK_ROOT)
+    set(OVR_SDK_ROOT "${OVR_SDK_ROOT}/OculusSDK")
+
+    mark_as_advanced(OVR_SDK_ROOT)
+    message(STATUS "Found OculusSDK: ${OVR_SDK_ROOT}")
 endif()
 
+set(LIBOVR_ROOT ${OVR_SDK_ROOT}/LibOVR)
+mark_as_advanced(LIBOVR_ROOT)
+
 # find include directory
 find_path(OVR_INCLUDE_DIR NAMES OVR_CAPI.h HINTS ${LIBOVR_ROOT}/Include)
 
 if(WIN32)
     if(MSVC)
-        find_library(OVR_LIBRARY NAMES LibOVR HINTS ${LIBOVR_ROOT}/Lib/Windows)
+        if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+            # compiling for 64 bit
+            set(MSVC_ARCH "x64")
+        else()
+            # compiling for 32 bit
+            set(MSVC_ARCH "Win32")
+        endif()
+
+        # select the correct library folder matching MSVC version
+        if (MSVC_VERSION GREATER 1800)
+            set(MSVC_NAME "VS2015")
+        elseif (MSVC_VERSION GREATER 1700)
+            set(MSVC_NAME "VS2013")
+        elseif (MSVC_VERSION GREATER 1600)
+            set(MSVC_NAME "VS2012")
+        else()
+            set(MSVC_NAME "VS2010")
+        endif()
+
+        find_library(OVR_LIBRARY NAMES LibOVR HINTS "${LIBOVR_ROOT}/Lib/Windows/${MSVC_ARCH}/Release/${MSVC_NAME}")
+
+        unset(MSVC_ARCH)
+        unset(MSVC_NAME)
```

Comment:
I would suggest using prefixes to avoid conflicts with user variables -- so e.g. `_OVR_MSVC_NAME` or something like that, leading underscore suggesting that this is an internal variable.